### ISSUE TITLE
Should there be a "top" type, root of the class hierarchy

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/concerto#180]
Should there be a root of the type hierarchy, akin to `Object` in many object-oriented languages.

Currently each different kind of class has it's own root `Asset` for assets, `Event` for events, etc.

Having a root for the type hierarchy has a number of benefits, most notably when used in the context of a programming language using CTO (as Ergo is). See for instance the related Ergo issue https://github.com/accordproject/ergo/issues/420

A possible way to achieve this is to use one of the existing classes, most probably `Concept` as the top of the hierarchy.

A possible corresponding definition for the systems types would become (in addition to the changes already included in the `release-1.0` branch):
```
namespace system
abstract concept Concept {  }
abstract asset Asset extends Concept {  }
abstract transaction Transaction extends Concept { }
abstract event Event extends Concept { }
abstract participant Participant extends Concept {  }
```

**Comment by dselman** [comment]
I *think* we have addressed this now in #181 - do you agree?

**Comment by jeromesimeon** [comment]
I _think_ so too!

I am wondering how we decide that it is? Is there any "subtype(Type1, Type2)" call in the API. Or "instanceof(Value,Type)" call in the API?

**Comment by dselman** [comment]
These changes have been merged to the `release-1.0` branch.
